internal/analytics/analytics_test.go

```diff
@@ -0,0 +1,193 @@
+	"go.uber.org/mock/gomock"
+)
+
+func TestAnalytics(t *testing.T) {
+	t.Run("EmitEvent should not send event if disabled", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		mockClient := analytics_mocks.NewMockHTTPClient(ctrl)
+
+		analyticsService := analytics.NewAnalyticsWithClient("test-token", "http://localhost", mockClient, false)
+		analyticsService.Disable()
+		analyticsService.EmitEvent(analytics.TrackEvent{Event: "test_event"})
+	})
+
+	t.Run("EmitEvent should send event if enabled", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		mockClient := analytics_mocks.NewMockHTTPClient(ctrl)
+
+		mockClient.EXPECT().Post(gomock.Any(), gomock.Any(), gomock.Any()).Return(&http.Response{
+			StatusCode: http.StatusOK,
+			Body:       io.NopCloser(strings.NewReader("1")),
+		}, nil)
+
+		analyticsService := analytics.NewAnalyticsWithClient("test-token", "http://localhost", mockClient, false)
+		analyticsService.EmitEvent(analytics.TrackEvent{Event: "test_event"})
+	})
+
+	t.Run("EmitEvent should send the correct event in the body", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		mockClient := analytics_mocks.NewMockHTTPClient(ctrl)
+
+		event := analytics.TrackEvent{
+			Event: "specific_event",
+			Properties: map[string]interface{}{
+				"key": "value",
+			},
+		}
+
+		mockClient.EXPECT().Post("http://localhost/track", gomock.Any(), gomock.Any()).
+			DoAndReturn(func(_, _ string, body io.Reader) (*http.Response, error) {
+				bodyBytes, err := io.ReadAll(body)
+				if err != nil {
+					t.Fatalf("error reading body: %v", err)
+				}
+
+				var decodedEvents []analytics.TrackEvent
+				err = json.Unmarshal(bodyBytes, &decodedEvents)
+				if err != nil {
+					t.Fatalf("error unmarshalling body: %v", err)
+				}
+				if len(decodedEvents) != 1 {
+					t.Fatalf("expected 1 event, got %d", len(decodedEvents))
+				}
+				decodedEvent := decodedEvents[0]
+
+				if decodedEvent.Event != "specific_event" {
+					t.Errorf("expected event 'specific_event', got '%s'", decodedEvent.Event)
+				}
+				properties, ok := decodedEvent.Properties.(map[string]interface{})
+				if !ok {
+					t.Fatalf("properties is not a map[string]interface{}")
+				}
+				if properties["key"] != "value" {
+					t.Errorf("expected properties['key'] to be 'value', got '%v'", properties["key"])
+				}
+
+				return &http.Response{
+					StatusCode: http.StatusOK,
+					Body:       io.NopCloser(strings.NewReader("1")),
+				}, nil
+			})
+
+		analyticsService := analytics.NewAnalyticsWithClient("test-token", "http://localhost", mockClient, false)
+		analyticsService.EmitEvent(event)
+	})
+}
+
+func TestEventCreation(t *testing.T) {
+	analyticsService := analytics.NewAnalyticsWithClient("test-token", "http://localhost", nil, false)
+
+	t.Run("NewGDSProjCreatedEvent", func(t *testing.T) {
+		event := analyticsService.NewGDSProjCreatedEvent()
+		if event.Event != "MCP4NEO4J_GDS_PROJ_CREATED" {
```

Comment:
where are we defining these event names?

Reply:
Inside the "events.go", 
eventually we could move these to be constants, but as I don't foresee I case for these to be reused, I think it improve the readability for the next guy having inline when they are needed, instead of create a layer of indirection IMO, no strongly opinionated to be fair, but I slightly prefer this way